db/versions/0009.yml

```diff
@@ -0,0 +1,288 @@
+            (cache_purges_entities_scan.rk is null or cache_purges_entities_scan.rk = cache_name)
+          order by cache_purges.provisioner_id, cache_purges.worker_type, cache_purges.cache_name
+          limit case
+            when (size is not null and size > 0) then size + 1
+            else null
+          end
+          offset case
+            when (size is not null and size > 0 and page is not null and page > 0) then page
+            else 0
+          end;
+        end if;
+      end
+  purge_requests:
+    description: |-
+      List the caches for this `provisioner_id_in`/`worker_type_in`.
+    mode: read
+    serviceName: purge_cache
+    args: provisioner_id_in text, worker_type_in text
+    returns: table (provisioner_id text, worker_type text, cache_name text, before timestamptz)
+    body: |-
+      begin
+        return query
+        select cache_purges.provisioner_id, cache_purges.worker_type, cache_purges.cache_name, cache_purges.before from cache_purges
+        where cache_purges.provisioner_id = provisioner_id_in and cache_purges.worker_type = worker_type_in;
+      end
+  all_purge_requests:
+    description: |-
+      View all active purge requests.
+    mode: read
+    serviceName: purge_cache
+    args: page_size_in integer, page_offset_in integer
+    returns: table (provisioner_id text, worker_type text, cache_name text, before timestamptz)
+    body: |-
+      declare
+      sql text := 'select cache_purges.provisioner_id, cache_purges.worker_type, cache_purges.cache_name, cache_purges.before from cache_purges';
+      begin
+        return query select cache_purges.provisioner_id, cache_purges.worker_type, cache_purges.cache_name, cache_purges.before
+        from cache_purges
+        order by cache_purges.provisioner_id, cache_purges.worker_type, cache_purges.cache_name
+        limit get_page_limit(page_size_in)
+        offset get_page_offset(page_offset_in);
+      end
+  purge_cache:
+    serviceName: purge_cache
+    description: |-
+      Publish a request to purge caches with name `cache_name_in`
+      on `provisioner_id_in`/`worker_type_in` workers.
+    mode: write
+    args: provisioner_id_in text, worker_type_in text, cache_name_in text, before_in timestamptz, expires_in timestamptz
+    returns: void
+    body: |-
+      declare
+        new_etag uuid := public.gen_random_uuid();
+      begin
+        insert into cache_purges(provisioner_id, worker_type, cache_name, before, expires, etag)
+        values (
+          provisioner_id_in,
+          worker_type_in,
+          cache_name_in,
+          before_in,
+          expires_in,
+          new_etag
+        ) on conflict (provisioner_id, worker_type, cache_name) do
+        update
+        set (before, expires, etag) = (before_in, expires_in, new_etag)
+        where cache_purges.provisioner_id = provisioner_id_in and cache_purges.worker_type = worker_type_in and cache_purges.cache_name = cache_name_in;
+      end
+  cache_purges_expires:
```

Comment:
Should this be something more like `expire_purge_caches`?

Reply:
#2754 (I checked with @helfi92..)